Remove rounding_mode="true" option

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50438 Support JIT shape analysis for div with rounding_mode argument
* #51989 Sparse support for division rounding_mode argument
* #51792 Vectorize floating point floor_divide
* #51988 Replace rounding_mode="true" with rounding_mode=None
* **#52139 Remove rounding_mode="true" option**

Differential Revision: [D26402844](https://our.internmc.facebook.com/intern/diff/D26402844)

peterbell10 committed:
Remove rounding_mode="true" option

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove rounding_mode="true" option"




[ghstack-poisoned]

mruberry commented on test/test_binary_ufuncs.py:
```diff
@@ -375,7 +375,7 @@ def test_div_rounding_numpy(self, device, dtype):
                 expect = torch.from_numpy(np_ref(an, bn))
 
-            kwargs = dict(rounding_mode=rounding_mode) if rounding_mode is not None else {}
+            kwargs = dict(rounding_mode=mode) if mode is not None else {}
```
Doesn't line 371 above need to be updated to be (None, np.true_divide)?

mruberry replied:
How did this test even work previously?

mruberry commented on test/test_binary_ufuncs.py:
```diff
@@ -327,7 +327,7 @@ def test_div_rounding_nonfinite(self, device, dtype):
             with np.errstate(all='ignore'):
                 expect = np_ref(an, bn)
-            kwargs = dict(rounding_mode=rounding_mode) if rounding_mode is not None else {}
+            kwargs = dict(rounding_mode=mode) if mode is not None else {}
```
See analogous comments below

mruberry commented on torch/_torch_docs.py:
```diff
@@ -2764,7 +2764,7 @@ def merge_dicts(*dicts):
     rounding_mode (str, optional): Type of rounding applied to the result:
 
-        * ``"true"`` - default behavior. Performs no rounding and, if both :attr:`input` and
+        * If not specified, performs no rounding and, if both :attr:`input` and
```
The signature on line 2744 needs to be updated to "round_mode=None, out=None". It currently still uses 'true' and there's no comma between the kwargs.

mruberry replied:
Divide signature also needs an update:

https://pytorch.org/docs/master/generated/torch.divide.html?highlight=rounding_mode

And true divide doc does, too: 

https://pytorch.org/docs/master/generated/torch.true_divide.html?highlight=rounding_mode

mruberry replied:
Method docs for div, div_, divide, and divide_ also need an update:

https://pytorch.org/docs/master/tensors.html?highlight=tensor%20div#torch.Tensor.div

peterbell10 committed:
Update on "Remove rounding_mode="true" option"




[ghstack-poisoned]

mruberry reviewed (commented):
Still some more cleanup to do. I made a few notes inline. There's also

https://github.com/pytorch/pytorch/blob/33afb5f19f4e427f099653139ae45b661b8bc596/torch/overrides.py#L371

and a couple places in the autograd computation, like here:

https://github.com/pytorch/pytorch/blob/fa325d7c9f167df898eee6be93ae1a8395525a79/torch/csrc/autograd/FunctionsManual.cpp#L310

peterbell10 committed:
Update on "Remove rounding_mode="true" option"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove rounding_mode="true" option"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove rounding_mode="true" option"




[ghstack-poisoned]

peterbell10 commented:
@mruberry I think I got all mentions of true this time. Also realised I needed to update the type hints since they mention "true".

peterbell10 committed:
Update on "Remove rounding_mode="true" option"


Differential Revision: [D26402844](https://our.internmc.facebook.com/intern/diff/D26402844)

[ghstack-poisoned]

peterbell10 commented:
FYI @mruberry I fixed a mistake in the type hints. Also XLA is still failing, but this time it compiles fine but errors when they pass `rounding_mode="true"`. So, it's an expected failure.

mruberry commented:
> FYI @mruberry I fixed a mistake in the type hints. Also XLA is still failing, but this time it compiles fine but errors when they pass `rounding_mode="true"`. So, it's an expected failure.

Got it, thanks for the update.
